fw/tls.c

```diff
@@ -398,16 +412,8 @@ tfw_tls_encrypt(struct sock *sk, struct sk_buff *skb, unsigned int mss_now,
 	 */
 	tfw_tcp_propagate_dseq(sk, skb_tail);
 	tcp_sk(sk)->write_seq += head_sz + TTLS_TAG_LEN;
-
-	/*
-	 * TLS record header is always allocated from the reserved skb headroom.
-	 * The room for the tag may also be allocated from the reserved tailroom
-	 * or in a new page fragment in skb_tail or next, probably new, skb.
-	 * So to adjust the socket write memory we have to check the both skbs
-	 * and only for TTLS_TAG_LEN.
-	 */
-	if (ss_add_overhead(sk, t_sz))
-		return -ENOMEM;
+	ss_del_overhead(sk, TLS_MAX_OVERHEAD - t_sz);
```

Comment:
So we call the socket memory allocation routines twice instead of once and actually the routines aren't so small - there is a lot of things happening inside them.